Source/WebGPU/WebGPU/RenderBundleEncoder.h

```diff
@@ -140,7 +140,8 @@ class RenderBundleEncoder : public WGPURenderBundleEncoderImpl, public RefCounte
     Vector<BufferAndOffset> m_fragmentBuffers;
     const Ref<Device> m_device;
     const RenderPipeline* m_pipeline { nullptr };
-    HashMap<uint32_t, Vector<uint32_t>, DefaultHash<uint32_t>, WTF::UnsignedWithZeroKeyHashTraits<uint32_t>> m_bindGroupDynamicOffsets;
+    using BindGroupDynamicOffsetsContainer = HashMap<uint32_t, Vector<uint32_t>, DefaultHash<uint32_t>, WTF::UnsignedWithZeroKeyHashTraits<uint32_t>>;
+    std::optional<BindGroupDynamicOffsetsContainer> m_bindGroupDynamicOffsets;
```

Comment:
Is an empty `BindGroupDynamicOffsetsContainer` equivalent to the "None" case of `std::optional`? (ie. I don't think the `std::optional` is required)

Reply:
I'm using std::optional to avoid building the dynamic offsets vertex and fragment buffers more than once.

`RenderBundleEncoder::replayCommands` sets `m_bindGroupDynamicOffsets = std::nullopt;` such that `m_dynamicOffsetsVertexBuffer` and `m_dynamicOffsetsFragmentBuffer` are not continuously rebuilt in `RenderBundleEncoder::executePreDrawCommands` since the contents can't change.

I understand this might be confusing as typing up that explanation was tricky to word correctly, so I am open to alternatives to achieving this